modules/programs/neomutt.nix

```diff
@@ -0,0 +1,325 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.neomutt;
+
+  sidebarModule = types.submodule ({ config, ...}: {
+    options = {
+      enable = mkEnableOption "sidebar support";
+
+      width = mkOption {
+        type = types.int;
+        default = 22;
+        description = "Width of the sidebar";
+      };
+
+      shortPath = mkOption {
+        type = types.bool;
+        default = true;
+        description = "By default sidebar shows the full path of the mailbox, but with this enabled only the relative name is shown.";
+      };
+
+      format = mkOption {
+        type = types.string;
+        default = "%B%?F? [%F]?%* %?N?%N/?%S";
+        description = "Sidebar format. Check neomutt documentation for details.";
+      };
+    };
+  });
+
+  bindModule = types.submodule ({ config, ...}: {
+    options = {
+      map = mkOption {
+        type = types.enum [
+          "alias"
+          "attach"
+          "browser"
+          "compose"
+          "editor"
+          "generic"
+          "index"
+          "mix"
+          "pager"
+          "pgp"
+          "postpone"
+          "query"
+          "smime"
+        ];
+        default = "index";
+        description = "Select the menu to bind the command to.";
+      };
+
+      key = mkOption {
+        type = types.string;
+        example = "<left>";
+        description = "The key to bind.";
+      };
+
+      action = mkOption {
+        type = types.string;
+        example = "<enter-command>toggle sidebar_visible<enter><refresh>";
+        description = "Specify the action to take.";
+      };
+    };
+  });
+
+  imapModule = types.submodule ({ config, ...}: {
+    options = {
+      enable = mkEnableOption "the IMAP support";
+
+      user = mkOption {
+        type = types.string;
+        default = "";
+        description = "IMAP username.";
+      };
+
+      password = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        example = "`${pkgs.gnupg}/bin/gpg --decrypt passfile.gpg`";
+        description = "IMAP password. Because these configuration files are kept in the nix store unencrypted, I recommend using some kind of a password command.";
```

Comment:
Please avoid such long lines. I much prefer something along the lines of
```
        description = ''
          IMAP password. Because these configuration files are kept in
          the nix store unencrypted, I recommend using some kind of a
          password command.
        '';
```
Also, instead of "I recommend" maybe "it is recommended".